extensions/iceberg/src/main/java/io/deephaven/iceberg/layout/IcebergFlatLayout.java

```diff
@@ -0,0 +1,80 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.iceberg.layout;
+
+import io.deephaven.base.FileUtils;
+import io.deephaven.engine.table.impl.locations.TableDataException;
+import io.deephaven.engine.table.impl.locations.impl.TableLocationKeyFinder;
+import io.deephaven.iceberg.location.IcebergTableLocationKey;
+import org.apache.iceberg.*;
+import org.apache.iceberg.io.FileIO;
+import org.jetbrains.annotations.NotNull;
+
+import java.net.URI;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Consumer;
+
+/**
+ * Iceberg {@link TableLocationKeyFinder location finder} for tables without partitions that will discover data files
+ * from a {@link org.apache.iceberg.Snapshot}
+ */
+public final class IcebergFlatLayout implements TableLocationKeyFinder<IcebergTableLocationKey> {
+    private final Snapshot tableSnapshot;
+    private final FileIO fileIO;
+    private final Map<URI, IcebergTableLocationKey> cache;
+    private final Object readInstructions;
+
+    private static IcebergTableLocationKey locationKey(
+            final FileFormat format,
+            final URI fileUri,
+            @NotNull final Object readInstructions) {
+        return new IcebergTableLocationKey(format, fileUri, 0, null, readInstructions);
+    }
+
+    /**
+     * @param tableSnapshot The {@link Snapshot} from which to discover data files.
+     * @param readInstructions the instructions for customizations while reading.
+     */
+    public IcebergFlatLayout(
+            @NotNull final Snapshot tableSnapshot,
+            @NotNull final FileIO fileIO,
+            @NotNull final Object readInstructions) {
+        this.tableSnapshot = tableSnapshot;
+        this.fileIO = fileIO;
+        this.readInstructions = readInstructions;
+
+        this.cache = new HashMap<>();
+    }
+
+    public String toString() {
+        return IcebergFlatLayout.class.getSimpleName() + '[' + tableSnapshot + ']';
```

Comment:
Should probably be table name. Should only include snapshot if we're not refreshing.